modules/plugin/jdbc/jdbc-postgis/src/test/java/org/geotools/data/postgis/PostGISTestSetup.java

```diff
@@ -50,11 +51,16 @@ public boolean isPgsqlVersionGreaterThanEqualTo(Version v) {
     protected void setUpDataStore(JDBCDataStore dataStore) {
         super.setUpDataStore(dataStore);
 
-        // the unit tests assume a non loose behaviour
-        ((PostGISDialect) dataStore.getSQLDialect()).setLooseBBOXEnabled(false);
-
-        // the tests assume non estimated extents
-        ((PostGISDialect) dataStore.getSQLDialect()).setEstimatedExtentsEnabled(false);
+        SQLDialect sqlDialect = dataStore.getSQLDialect();
+        if (sqlDialect instanceof PostGISDialect) {
+            PostGISDialect d = (PostGISDialect) sqlDialect;
+            // the unit tests assume a non loose behaviour
+            d.setLooseBBOXEnabled(false);
+            d.setEstimatedExtentsEnabled(false);
+        } else {
+            PostGISPSDialect d = (PostGISPSDialect) sqlDialect;
+            d.setLooseBBOXEnabled(false);
+        }
```

Comment:
This change appears to be unrelated, can you describe why it was done?

Reply:
It was not unrelated. But I found a better solution which is more limited to the affected tests. I will push that instead.